### ISSUE TITLE
gmake fail: "No rule to make target /usr/lib46/libboost_context-mt.so"

### ISSUE THREAD

**Issue by tmsimont** [STEllAR-GROUP/hpx#2022]
I'm trying to build the project from source on Linux.

I'm following the instructions in the README

I got through the first 4 steps, just passed this:

```
cmake -DBOOST_ROOT=/usr/lib64 -DHWLOC_ROOT=/usr/lib64 -DTCMALLOC_INCLUDE_DIR=/usr/lib64 -DTCMALLOC_LIBRARY=libtcmalloc.so.4 ../
```

I then ran
```
gmake -j5
```

(four core machine)

And gmake fails 50% in:

```
[ 50%] Building CXX object src/CMakeFiles/hpx.dir/__/plugins/parcelport/tcp/conn      ection_handler_tcp.cpp.o
gmake[2]: *** No rule to make target '/usr/lib64/libboost_context-mt.so', needed       by 'lib/libhpx.so.0.9.12'.  Stop.
gmake[2]: *** Waiting for unfinished jobs....
[ 50%] Building CXX object src/CMakeFiles/hpx.dir/__/plugins/parcelport/tcp/parc      elport_tcp.cpp.o
CMakeFiles/Makefile2:1626: recipe for target 'src/CMakeFiles/hpx.dir/all' failed
gmake[1]: *** [src/CMakeFiles/hpx.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
```

I checked for `/usr/lib64/libboost_context-mt.so` and the file is there. I'm wondering -- what does this message mean? How can I go about resolving this?

**Comment by zao** [comment]
Ensure that it's not a broken symlink, some distros install the convenience symlinks but not the actual versioned library file.

**Comment by tmsimont** [comment]
Good call. Looks like it is in fact a broken symlink...

**Comment by tmsimont** [comment]
Dang now I'm getting a different error:

```
[ 50%] gmake[2]: *** No rule to make target 'libtcmalloc.so.4.2.1', needed by 'lib/libhpx.so.0.9.12'.  Stop.
```

That one is definitely not a broken symlink:
```
ls -al /usr/lib64/libtcmalloc.so.4.2.1
-rwxr-xr-x 1 root root 309568 Oct  6  2014 /usr/lib64/libtcmalloc.so.4.2.1
```

**Comment by hkaiser** [comment]
Please give us the full configure and build logs.

**Comment by tmsimont** [comment]
Interesting... 

I tried to just add a symlink:

`ln -s /usr/lib64/libtcmalloc.so.4 /usr/lib64/libtcmalloc.so`

I re-ran the cmake and now it works!

I should note that I needed to manually enter either `TCMALLOC_LIBRARY` or `TCMALLOC_INCLUDE_DIR`  (via `cmake ... -DTCMALLOC_INCLUDE_DIR=...`) in order for Cmake to work in the first place.

Entering either TCMALLOC_LIBRARY (`cmake ... -DTCMALLOC_LIBRARY=...`) by itself, or TCMALLOC_LIBRARY + TCMALLOC_INCLUDE_DIR (`cmake ... -DTCMALLOC_LIBRARY=... -DTCMALLOC_INCLUDE_DIR=...`) would get cmake to work, but produced the output:
`No rule to make target 'libtcmalloc.so.4.2.1'`
During the `gmake -j5` step.

Entering just TCMALLOC_INCLUDE_DIR  (`cmake ... -TCMALLOC_INCLUDE_DIR =...`) would cause a failure in CMake:
`Could NOT find TCMalloc`

It seems that by setting the include directory and the tmalloc library with cmake compiler arguments that CMake was able to find my `/usr/lib64/libtcmalloc.so.4.2.1` file, but when I went to `gmake -j5`, the `gmake` process failed to find that same .so, and the build fails. 

Apparently `gmake` can only find the library if it's named `libtcmalloc.so` (minus the .4.2.1 suffix), but `cmake` didn't have that problem...

In any case. I fixed it by simply adding a symlink.

Thank you for your help.
